### PR TITLE
Make `include` docs available at REPL again + be explicit about global scope

### PR DESCRIPTION
**Pull request by mauro3**
The docs of `include` were not accessible at the REPL anymore because it was moved to the `MainInclude` module.

Also, docs now explicitly state that it evaluates in the module's global scope, this comes up now and then: https://discourse.julialang.org/t/is-include-safe-to-use-inside-a-function/10867/4

### CODE REVIEW

**Comment by JeffBezanson** [comment]
`Base.include` accepts a module argument and should still be documented.

**Comment by mauro3** [comment]
I am a bit confused to where to attach the docs, as all those module local `include` are all different generic function. I think it would be best to make separate, slightly different, docs for `MainInclude.include` and `Base.include`, right?

Also, then the `@ref` to `include` here:
https://github.com/JuliaLang/julia/pull/27089/files#diff-a03d5ce5729a81495000698e643fce27R1093 probably should point to `Base.include`, right?

**Comment by mauro3** [comment]
I updated the PR.  Also, as a second commit, I updated the manual to explicitly state the `include` evaluates code in the global scope of the module and that each module includes its own definition of `include`.

**Comment by fredrikekre** [comment]
Do the docstrings who up in the manual with this change? xref  #22940

**Comment by mauro3** [comment]
You mean "...end up..."?  Just the `Base.include` goes to "Essentials".  To make the other appear I would have to add `MainInclude.include` to https://github.com/JuliaLang/julia/blob/master/doc/src/base/base.md#getting-around, which would be a bit odd.  Any idea on how to do this?

Also this PR should probably wait until #27098 settles to be sure to be consistent.

**Comment by JeffBezanson** [comment]
`include` is a bit more complex than `eval`. `Base.include(m, file)` adds extra stuff on top of `Core.include(m, file)`, so I'm not sure we can handle it the exact same way as `eval`.

File: doc/src/manual/modules.md [hunk 1]
```diff
@@ -145,6 +146,7 @@ using Base
 
 eval(x) = Core.eval(Mod, x)
 eval(m,x) = Core.eval(m, x)
+include(p) = Core.include(Mod, p)
```
Comment (JeffBezanson) [inline doc/src/manual/modules.md]: Actually `Base.include`.

**Comment by mauro3** [comment]
> Do the docstrings show up in the manual with this change?

Now, I also added the `Base.MainInclude.include` doc-string in the manual (as is also done in #27098).

Presumably the previous [Travis OSX failure](https://travis-ci.org/JuliaLang/julia/jobs/379068712) is unrelated.

**Pull request merged**